pkg/operator2/configobservation/oauth/template_conversions.go

```diff
@@ -0,0 +1,106 @@
+package oauth
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	corelistersv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	osinv1 "github.com/openshift/api/osin/v1"
+)
+
+const (
+	ocpBrand = "ocp"
+	okdBrand = "okd"
+)
+
+// ----- partial copy of console CLI config -----
+
+type ConsoleConfig struct {
+	Customization `yaml:"customization"`
+}
+
+type Customization struct {
+	Branding string `yaml:"branding"`
+}
+
+// ----- end of copy -----
+
+func convertTemplatesWithBranding(cmLister corelistersv1.ConfigMapLister, configTemplates *configv1.OAuthTemplates) (*osinv1.OAuthTemplates, map[string]string, error) {
+	templates := osinv1.OAuthTemplates{}
+	templateSyncData := map[string]string{}
+
+	brand, err := getConsoleBranding(cmLister)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	switch brand {
+	case okdBrand:
+		// do nothing, OAuth server has right branding for this
+
+	case ocpBrand, "dedicated", "online", "azure":
+		// all of these are equivalent to ocp
+		templates = osinv1.OAuthTemplates{
```

Comment:
Maybe define this outside of the switch statement and use it here and for default?